### PR TITLE
EBS support 

### PR DESCRIPTION
**Pull request by enxebre**
As part of CLOUD-303 this introduce support for BlockDevices

### CODE REVIEW

**Comment by enxebre** [comment]
cc @bison

File: pkg/cloud/aws/actuators/machine/instances.go [hunk 1]
```diff
@@ -250,6 +292,9 @@ func launchInstance(machine *clusterv1.Machine, machineProviderConfig *providerc
 		Placement:          placement,
 	}
 
+	if len(blockDeviceMappings) > 0 {
```
Comment (ingvagabund) [inline pkg/cloud/aws/actuators/machine/instances.go]: You can skip the check given `getBlockDeviceMappings` always returns valid list of block device mappings (including nil = empty array).

**Comment by ingvagabund** [comment]
/approve

would be great to provide example as well

File: pkg/actuators/machine/instaces_test.go [hunk 2]
```diff
@@ -90,6 +90,70 @@ func TestBuildEC2Filters(t *testing.T) {
...
+					VirtualName: nil,
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		got, err := getBlockDeviceMappings(tc.blockDevices, "existing-AMI", mockAWSClient)
+		if err != nil {
+			t.Errorf("error when calling getBlockDeviceMappings: %v", err)
```
Comment (frobware) [inline pkg/actuators/machine/instaces_test.go]: nit: should this be fatalf? can the next assertion reliably run in the face of an error?

Comment (enxebre) [reply pkg/actuators/machine/instaces_test.go]: Ideally yes, how ever keep in mind that the way it's right now as soon as any assertion returns error, the tests will stop as a trade off of choosing our custom implementation over oddly accommodating go test for running this e2e. We could collect errors in main before returning

Comment (frobware) [reply pkg/actuators/machine/instaces_test.go]: errorf will continue in the same test function.

**Pull request merged**